### PR TITLE
Fix Bug AB#7614 - Missing navigation properties

### PR DESCRIPTION
**Pull request by sslaws**
The model has been updated and defect 7615 was created as I had to temporarily add IgnoreDataMember so that serialization handled in the existing manner.  

In Drug Product, I added navigations to each child table.
I moved the DrugProduct keys in each child table to the end of the table. =

The relationships are now enforced and the Federal Drug files loaded correctly.

### CODE REVIEW

**Comment by healthgatewaybc** [comment]
Database Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=databaselibrary_pr_185)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_185)

File: Apps/Database/src/Models/Form.cs [hunk 1]
```diff
@@ -26,15 +26,24 @@ public class Form : AuditableEntity
...
         public string PharmaceuticalForm { get; set; }
 
         [MaxLength(80)]
         public string PharmaceuticalFormFrench { get; set; }
+
+        /// <summary>
+        /// Gets or sets the object Drug Product reference.
+        /// </summary>
+        [Required]
+        public virtual DrugProduct DrugProduct { get; set; }
```
Comment (FuriousLlama) [inline Apps/Database/src/Models/Form.cs]: if the drug product has a reference to this table, is it needed here?

Comment (sslaws) [reply Apps/Database/src/Models/Form.cs]: This is how foreign key relationships are enforced.

**Comment by healthgatewaybc** [comment]
Database Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=databaselibrary_pr_185)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_185)

**Comment by healthgatewaybc** [comment]
Database Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=databaselibrary_pr_185)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_185)

**Pull request merged**